src/components/Reports/SalaryReport/SalaryReportTable.tsx

```diff
@@ -1,10 +1,285 @@
+      </TableCell>
+    </TableRow>
+  );
+
+  const handlePageChange = (
+    event: React.MouseEvent<HTMLButtonElement> | null,
+    newPage: number,
+  ): void => {
+    setPage(newPage);
+  };
+
+  const handleLimitChange = (event: ChangeEvent<HTMLInputElement>): void => {
+    setLimit(parseInt(event.target.value));
+  };
+
+  const paginatedContacts = applyPagination(contacts, page, limit);
+
+  // let years: any[] = [];
+  // currencyGroups?.totals?.months.forEach((item: any) => {
+  //   const yearItem = { count: 1, year: item.split('-')[0] };
+  //   let found = false;
+  //   years = years.map((year) => {
+  //     if (year.year === yearItem.year) {
+  //       found = true;
+  //       return { count: year.count + 1, ...year };
+  //     }
+  //     return year;
+  //   });
+
+  //   if (!found) {
+  //     years.push(yearItem);
+  //   }
+  // });
+
+  return (
+    <Box>
+      <Box my={2}>
+        <Grid
+          container
+          justify="space-between"
+          className={clsx(classes.root, className)}
+          {...rest}
+        >
+          <Grid item>
+            <Typography variant="h5">{t(title)}</Typography>
+          </Grid>
+          <Grid item>
+            <ButtonGroup aria-label="report header button group">
+              <Button
+                startIcon={
+                  <SvgIcon fontSize="small">
+                    <CodeIcon />
+                  </SvgIcon>
+                }
+              >
+                Expand Partner Info
+              </Button>
+              <Button
+                startIcon={
+                  <SvgIcon fontSize="small">
+                    <GetAppIcon />
+                  </SvgIcon>
+                }
+              >
+                <CSVLink
+                  data={contacts}
+                  filename={`mpdx-salary-contributions-export-${DateTime.now().toISODate()}.csv`}
+                  className={classes.downloadCsv}
+                >
+                  Export
+                </CSVLink>
+              </Button>
+              <Button
+                startIcon={
+                  <SvgIcon fontSize="small">
+                    <PrintIcon />
+                  </SvgIcon>
+                }
+              >
+                Print
+              </Button>
+            </ButtonGroup>
+          </Grid>
+        </Grid>
+      </Box>
+      <TableContainer>
+        <Table>
+          <TableHead>
+            <TableRow>
+              <TableCell>
+                <Typography variant="h6">{salaryCurrency}</Typography>
+              </TableCell>
+              {currencyGroups?.map((currencyGroup) => {
+                const allYears = currencyGroup.totals.months.map(
+                  (month) => month.month.split('-')[0],
+                );
+                const monthCount = allYears.reduce(
+                  (year, count) => (
+                    (year[count] = (year[count] || 0) + 1), year
```

Comment:
This is a comma expression right? I think it's a little hard to read. We could use spread instead:
```suggestion
                    {
                      ...year,
                      [count]: (year[count] || 0) + 1,
                    } 
```